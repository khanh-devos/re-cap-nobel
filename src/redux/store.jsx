import { configureStore } from '@reduxjs/toolkit';
import NobelReducer from './nationSlice/NationSlice';

const store = configureStore({
  reducer: {
    nobel: NobelReducer,
  },
});

export default store;
