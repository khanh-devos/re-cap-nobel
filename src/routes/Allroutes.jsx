import { useSelector } from 'react-redux';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { v4 as uuidv4 } from 'uuid';

import Nation from '../components/Nation';
import CityRoute from './CityRoute';
import NationRoute from './NationRoute';

export default function Allroutes() {
  const { uniqueCountries } = useSelector((store) => store.nobel);

  return (
    <BrowserRouter basename="re-cap-nobel">
      <Routes>
        <Route path="/" element={<Nation />} />
        {
            uniqueCountries.map(({ id, country, cities }) => {
              const r1 = [
                <Route
                  key={uuidv4()}
                  path={`/${country}`}
                  element={<NationRoute country={country} />}
                />,
              ];

              cities.forEach((city) => (
                r1.push(<Route
                  key={uuidv4()}
                  path={`/${country}/${city}`}
                  element={<CityRoute city={city} country={country} />}
                />)));

              return r1;
            })
        }
      </Routes>
    </BrowserRouter>
  );
}
