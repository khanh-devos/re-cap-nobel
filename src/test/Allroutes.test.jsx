import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Allroutes from '../routes/Allroutes';

it('should match the snapshot Allroutes', async () => {
  const { container } = render(
    <Provider store={store}>
      <Allroutes />
    </Provider>,
  );

  expect(container).toMatchSnapshot();
});
