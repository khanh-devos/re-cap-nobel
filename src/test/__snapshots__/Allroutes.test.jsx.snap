// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot Allroutes 1`] = `<div />`;
